src/idl/include/idl/processor.h

```diff
@@ -125,10 +126,19 @@ struct idl_builtin_annotation {
   const idl_annotation_callback_t callback;
 };
 
+typedef struct idl_config idl_config_t;
+struct idl_config {
+  const char *working_directory;
+  const char *path; /**< <stdin>, <string> or pathname */
```

Comment:
So what's going to happen if I create a file `<stdin>`?

Answer: I'll get `a.c` and `a.h` instead of `<stdin>.c` and `<stdin>.h` 😢 A contrived case, admittedly, but I think it would not be harder to define an enum for file/stdin/string and then you'll be protected from such annoying people.